flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/MergeTableLikeUtil.java

```diff
@@ -426,6 +433,14 @@ private void appendDerivedColumns(
                         }
                     }
 
+                    Set<String> physicalFieldNames = physicalFieldNamesToTypes.keySet();
+                    Set<String> metadataFieldNames = metadataFieldNamesToTypes.keySet();
+                    final Set<String> result = new LinkedHashSet<>(physicalFieldNames);
+                    result.retainAll(metadataFieldNames);
+                    if (!result.isEmpty()) {
+                        throw new ValidationException(
+                                "A field name conflict exists between a field of the regular type and a field of the Metadata type.");
+                    }
```

Comment:
may be we can just check duplication when put the new Column to the columns, at the end of this function?

Reply:
> may be we can just check duplication when put the new Column to the columns, at the end of this function?

hi @wenlong88 
Thanks for your review and comments, I think we can adjust the determination of whether a SqlRegularColumn type field is duplicated with other types of fields to check if a new column is duplicated when put into the column at the end of this function.
But accessibleFieldNamesToTypes heavy logic to this place, I think it is necessary, because, when creating computecolcolumn if metadata columns in the former, repetitive physical column in the bottom, In the use of accessibleFieldNamesToTypes putAll, metadata list field covers physical column, if the user want, according to the calculated regularColumn you don't need to do so.

Reply:
> may be we can just check duplication when put the new Column to the columns, at the end of this function?

hi @wenlong88 
Thanks for your review and comments, I think we can adjust the determination of whether a SqlRegularColumn type field is duplicated with other types of fields to check if a new column is duplicated when put into the column at the end of this function.
But accessibleFieldNamesToTypes heavy logic to this place, I think it is necessary, because, when creating computecolcolumn if metadata columns in the former, repetitive physical column in the bottom, In the use of accessibleFieldNamesToTypes putAll, metadata list field covers physical column, if the user want, according to the calculated regularColumn you don't need to do so.

Reply:
> may be we can just check duplication when put the new Column to the columns, at the end of this function?

hi,@wenlong88 Do you think my view is correct?

Reply:
hi, @jelly-1203 thanks for the update and  analysis, I agree that it not  possible to add unified check at the end. but I still think that the check here can be improved a bit:
1. it seems that the check here is not relevant to the new added ComputingColumn, if you want to check the duplication, it is better to add it when updating metadataFieldNamesToTypes.
2. according to current implementation, regular columns have top priority(we collect collectPhysicalFieldsTypes at the beginning), so we may also need to check if there is duplicated name in physicalFieldNamesToTypes when trying to a metadata column or computed column. If we add such check, the check in 1 is not necessary any more.

what do you think?

Reply:
> hi, @jelly-1203 thanks for the update and analysis, I agree that it not possible to add unified check at the end. but I still think that the check here can be improved a bit:
> 
> 1. it seems that the check here is not relevant to the new added ComputingColumn, if you want to check the duplication, it is better to add it when updating metadataFieldNamesToTypes.
> 2. according to current implementation, regular columns have top priority(we collect collectPhysicalFieldsTypes at the beginning), so we may also need to check if there is duplicated name in physicalFieldNamesToTypes when trying to a metadata column or computed column. If we add such check, the check in 1 is not necessary any more.
> 
> what do you think?

hi, @wenlong88 I think your suggestion is reasonable, I will try to modify it